frontend/src/features/public-form/PublicFormProvider.tsx

```diff
@@ -494,7 +495,9 @@ export const PublicFormProvider = ({
 
   const isPaymentEnabled =
     data?.form.responseMode === FormResponseMode.Encrypt &&
-    data.form.payments_field.enabled
+    data.form.payments_channel.channel !== PaymentChannel.Unconnected &&
+    !!data.form.payments_field.amount_cents &&
```

Comment:
If `payments_field` children are required, then over here we can just check if `payments_field` exists

Reply:
As mentioned above, we can't cos payments_field children can't be required?